code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm

```diff
@@ -353,4 +359,215 @@
+
+/mob/living/simple_animal/hostile/jungle/mook/proc/WarmupAttack(forced_slash_combo = FALSE)
+	if(attack_state == MOOK_ATTACK_NEUTRAL && target)
+		attack_state = MOOK_ATTACK_WARMUP
+		notransform = TRUE
+		walk(src,0)
+		update_icons()
+		if(prob(50) && get_dist(src,target) <= 3 || forced_slash_combo)
+			addtimer(CALLBACK(src, .proc/SlashCombo), ATTACK_INTERMISSION_TIME)
+			return
+		addtimer(CALLBACK(src, .proc/LeapAttack), ATTACK_INTERMISSION_TIME + rand(0,3))
+		return
+	attack_state = MOOK_ATTACK_RECOVERY
+	ResetNeutral()
+
+/mob/living/simple_animal/hostile/jungle/mook/proc/SlashCombo()
+	if(attack_state == MOOK_ATTACK_WARMUP && !stat)
+		attack_state = MOOK_ATTACK_ACTIVE
+		update_icons()
+		SlashAttack()
+		addtimer(CALLBACK(src, .proc/SlashAttack), 3)
+		addtimer(CALLBACK(src, .proc/SlashAttack), 6)
+		addtimer(CALLBACK(src, .proc/AttackRecovery), 9)
+
+/mob/living/simple_animal/hostile/jungle/mook/proc/SlashAttack()
+	if(target && !stat && attack_state == MOOK_ATTACK_ACTIVE)
+		melee_damage_lower = 15
+		melee_damage_upper = 15
+		var/mob_direction = get_dir(src,target)
+		if(get_dist(src,target) > 1)
+			step(src,mob_direction)
+		if(targets_from && isturf(targets_from.loc) && target.Adjacent(targets_from) && isliving(target))
+			var/mob/living/L = target
+			L.attack_animal(src)
+			return
+		var/swing_turf = get_step(src,mob_direction)
+		new /obj/effect/temp_visual/kinetic_blast(swing_turf)
+		playsound(src, 'sound/weapons/slashmiss.ogg', 50, 1)
+
+/mob/living/simple_animal/hostile/jungle/mook/proc/LeapAttack()
+	if(target && !stat && attack_state == MOOK_ATTACK_WARMUP)
+		attack_state = MOOK_ATTACK_ACTIVE
+		density = FALSE
+		melee_damage_lower = 30
+		melee_damage_upper = 30
+		update_icons()
+		new /obj/effect/temp_visual/mook_dust(get_turf(src))
+		playsound(src, 'sound/weapons/thudswoosh.ogg', 25, 1)
+		playsound(src, 'sound/voice/mook_leap_yell.ogg', 100, 1)
+		var/target_turf = get_turf(target)
+		throw_at(target_turf, 7, 1, src, FALSE, callback = CALLBACK(src, .proc/AttackRecovery))
+		return
+	attack_state = MOOK_ATTACK_RECOVERY
+	ResetNeutral()
+
+/mob/living/simple_animal/hostile/jungle/mook/proc/AttackRecovery()
+	if(attack_state == MOOK_ATTACK_ACTIVE && !stat)
+		attack_state = MOOK_ATTACK_RECOVERY
+		density = TRUE
+		face_atom(target)
+		if(!struck_target_leap)
+			update_icons()
+		struck_target_leap = FALSE
+		if(prob(40) && !ckey)
+			attack_state = MOOK_ATTACK_NEUTRAL
+			if(target)
+				if(isliving(target))
+					var/mob/living/L = target
+					if(L.incapacitated() && L.stat < 2)
+						addtimer(CALLBACK(src, .proc/WarmupAttack, TRUE), ATTACK_INTERMISSION_TIME)
+						return
+			addtimer(CALLBACK(src, .proc/WarmupAttack), ATTACK_INTERMISSION_TIME)
+			return
+		addtimer(CALLBACK(src, .proc/ResetNeutral), ATTACK_INTERMISSION_TIME)
+
+/mob/living/simple_animal/hostile/jungle/mook/proc/ResetNeutral()
+	if(attack_state == MOOK_ATTACK_RECOVERY)
+		attack_state = MOOK_ATTACK_NEUTRAL
+		notransform = FALSE
```

Comment:
that would be ideal; a better fix would be to make simple animals respect being stunned, and thus apply a stun that ignores their lack of CANSTUN flag